source/common/common/fine_grain_logger.cc

```diff
@@ -73,10 +71,9 @@ void FineGrainLogContext::setDefaultFineGrainLogLevelFormat(spdlog::level::level
     ABSL_LOCKS_EXCLUDED(fine_grain_log_lock_) {
   absl::WriterMutexLock wl(&fine_grain_log_lock_);
   verbosity_default_level_ = level;
-  verbosity_update_info_.clear();
-  for (const auto& it : *fine_grain_log_map_) {
-    it.second->set_level(level);
-    it.second->set_pattern(format);
+  for (auto& [key, logger] : *fine_grain_log_map_) {
+    logger->set_level(getLogLevel(key));
+    logger->set_pattern(format);
```

Comment:
if verbosity_update_info is empty, we don't need to query any of these levels, and can simply set them.

Reply:
The current admin server interface is using the `setAllFineGrainLoggers()`, which is stateless and may make `FinGrainLogContext` inconsistent. It needs to be migrated to the verbosity default level update. We can refactor and generalize the update part if the above doesn't sound good, eg.
```
void FineGrainLogContext::updateAllFineGrainLoggers()
    ABSL_LOCKS_EXCLUDED(fine_grain_log_lock_) {
  absl::ReaderMutexLock l(&fine_grain_log_lock_);
  if (verbosity_update_info_.empty()) {
    for (const auto& it : *fine_grain_log_map_) {
      it.second->set_level(verbosity_default_level_);
    }
  } else {
    for (const auto& [key, logger] : *fine_grain_log_map_) {
      logger->set_level(getLogLevel(key));
    }
  }
}
```

`setDefaultFineGrainLogLevelFormat` includes format setting, which will only be called during initialization, and we can leave as it is, I think.

Reply:
Well, as a post request in the admin hook, it makes sense to be stateless as well... We can discuss more for the admin hook. These are now enough for the google3 hook.

Reply:
Let's go with the second approach as it's slightly better as we'd then have more consistent behavior with the other system. As the admin doesn't yet have a hook to set the verbosity_update_info it'd fall back to behaving as is now -- this will just help clients who can use verbosity_update_info to not run into inconsistencies.

Reply:
Done.